recipes/gtk/all/conanfile.py

```diff
@@ -87,6 +88,12 @@ def validate(self):
         if self.settings.os == "Linux" and (self.options.with_wayland or self.options.with_x11) and not self.options["pango"].with_freetype:
             raise ConanInvalidConfiguration(
                 "gtk requires pango with freetype when built with wayland/x11 support")
+        if self.settings.compiler == "clang":
```

Comment:
This would prevent a user from building the recipe locally using their own installed glibc (which works).

Reply:
I'm not sure I get your point. What I'm trying to prevent is mixing different versions of "glib" (not glibc) which I found to cause problems.

Reply:
The recipe can be successfully built from source with clang using `conan create --build`. If you make clang an unsupported configuration, this will not be possible.

Reply:
In which case the solution would be to require shared glib by default rather than disabling clang entirely.

Reply:
We shouldn't mix different versions of glib. Gdk-pixbuf couldn't be built using the newer versions of glib. it's not clear to me that it would behave correctly if it is built and linked against an old glib, and then at runtime loaded against newer glib.

Even if it's possilbe, it wouldn't work, because I made dependencies of glib use conan's "full_package_mode" to enforce using the same version of glib (because mixing versions is problematic).

Reply:
Right, but you can achieve that with
```
conan create gtk/4.70@ --build
```
or
```
conan create gtk/4.70.0 -o glib:shared=True --build
```

I guess what I'm saying is that disabling clang entirely at the recipe level seems like a drastic change that will have a high impact on a large number of users, when there's a workaround available either from the command line or by setting a build policy in downstream recipes.